cms/djangoapps/contentstore/views/videos.py

```diff
@@ -291,12 +291,17 @@ def validate_transcript_preferences(provider, cielo24_fidelity, cielo24_turnarou
 
                 # Validate transcription languages
                 supported_languages = transcription_plans[provider]['fidelity'][cielo24_fidelity]['languages']
+                if video_source_language not in supported_languages:
```

Comment:
@muzaffaryousaf modularity will be improved as a part of [EDUCATOR-1494](https://openedx.atlassian.net/browse/EDUCATOR-1494) eventually. 

So, keeping this change in the flow for now.

Reply:
Should we add a TODO for that?